test/interval.jl

```diff
@@ -887,4 +887,140 @@
             end
         end
     end
+
+    @testset "floor" begin
+        # `on` keyword is required
+        @test_throws UndefKeywordError floor(Interval(0.0, 1.0))
+
+        # only :left and :right are supported
+        @test_throws MethodError floor(Interval(0.0, 1.0); on=:nothing)
+
+        @test floor(Interval(0.0, 1.0); on=:left) == Interval(0.0, 1.0)
+        @test floor(Interval(0.5, 1.0); on=:left) == Interval(0.0, 0.5)
+        @test floor(Interval(0.0, 1.5); on=:left) == Interval(0.0, 1.5)
+        @test floor(Interval(0.5, 1.5); on=:left) == Interval(0.0, 1.0)
+
+        @test floor(Interval(0.0, 1.0); on=:right) == Interval(0.0, 1.0)
+        @test floor(Interval(0.5, 1.0); on=:right) == Interval(0.5, 1.0)
+        @test floor(Interval(0.0, 1.5); on=:right) == Interval(-0.5, 1.0)
+        @test floor(Interval(0.5, 1.5); on=:right) == Interval(0.0, 1.0)
+
+        # :anchor is only usable with AnchoredIntervals
+        @test_throws MethodError floor(Interval(0.0, 1.0); on=:anchor)
```

Comment:
Is it worth while to implement this method which then throws an explicit error saying you cannot anchor on a non-AnchoredInterval?

I ask cause it may seem at first glance like this is missing an implementation if you attempt it.